packages/xsnap/makefiles/lin/xsnap.mk

```diff
@@ -0,0 +1,139 @@
+% : %.c
+%.o : %.c
+
+GOAL ?= debug
+NAME = xsnap
+ifneq ($(VERBOSE),1)
+MAKEFLAGS += --silent
+endif
+
+# This Makefile gets invoked as "make" with this directory as the "cwd" using
+# Node.js child_process.spawn. For reasons that remain mysterious to this
+# author, under these circumstances, $(PWD) is two parent directories up
+# from this file. At the shell, (cd makefiles/lin && make) works fine.
+# Regardless, using HERE instead of PWD makes this Makefile work regardless of
+# the working directory.
+HERE = $(dir $(realpath $(firstword $(MAKEFILE_LIST))))
+MODDABLE = $(HERE)/../../moddable
+XS_DIR = $(MODDABLE)/xs
+BUILD_DIR = $(HERE)/../../build
+
+BIN_DIR = $(BUILD_DIR)/bin/lin/$(GOAL)
+INC_DIR = $(XS_DIR)/includes
+PLT_DIR = $(XS_DIR)/platforms
+SRC_DIR = $(XS_DIR)/sources
+TLS_DIR = ../../src
+TMP_DIR = $(BUILD_DIR)/tmp/lin/$(GOAL)/$(NAME)
+
+MACOS_ARCH ?= -arch i386
+MACOS_VERSION_MIN ?= -mmacosx-version-min=10.7
+
+C_OPTIONS = \
+	-fno-common \
+	-DINCLUDE_XSPLATFORM \
+	-DXSPLATFORM=\"xsnap.h\" \
+	-DmxDebug=1 \
```

Comment:
if `mxDebug` is the flag that either makes normal startup impossible (because intentional exceptions wait for an `xsbug` that isn't running) or makes debugging impossible (because exceptions have no stack traces), maybe we should paint an arrow/comment on it here.

Reply:
This `#define` causes some additional functions to be incorporated in the binary, but the DEBUG environment variable is also necessary and links in additional static libraries and reduces the optimization latitude given to `gcc`. I’m not sure why mxDebug is always set, but I assume for good reason.